src/convertToObject.js

```diff
@@ -6,7 +6,26 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
-}
+  let strToObject = sourceString
+    .trim()
+    .split(';')
+    .filter((str) => str.trim());
+
+  strToObject = strToObject.map((str) => {
+    return str
+      .trim()
+      .split(':')
+      .map((s) => s.trim());
+  });
+
+  strToObject = Object.fromEntries(strToObject);
 
+  for (const key in strToObject) {
```

Comment:
The use of a for-in loop to remove undefined properties is unnecessary. The way the code is structured, undefined properties should not occur. If they do, it's likely a symptom of a different issue that should be addressed directly.